### PR TITLE
Pass in label to support the shadow DOM

### PR DESCRIPTION
**Pull request by russcarver**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,30 @@ if (!supportsTime) {
...
 This will add a global `TimePolyfill` function to the page.
 
+### Shadow DOM
+
+When your code is inside a component that resides in the Shadow DOM, the polyfill may not be able to find your label
+element. For this case, you can pass your label element in directly.
+
+```
+<label id="myLabel"></label>
+<input #time type="time" id="timeInput">
```
Comment (Dan503) [inline README.md]: What is with the `#time` attribute? I don't think that is valid HTML.

File: README.md [hunk 2]
```diff
@@ -128,6 +128,30 @@ if (!supportsTime) {
...
+```         
+
+```               
+// This example is in Angular/Typscript
```
Comment (Dan503) [inline README.md]: Don't use angular or Typescript for the example. Anyone using those things will know how to translate the example. People who don't use those things will just get confused by the Typescript and Angular parts.

File: README.md [hunk 3]
```diff
@@ -128,6 +128,30 @@ if (!supportsTime) {
...
+
+import * as timePolyFill from 'time-input-polyfill';
+
+public constructor(private elemRef: ElementRef) {
```
Comment (Dan503) [inline README.md]: I want to display the simplest syntax possible.

The example shouldn't use class syntax.

Try to make the example as simple as possible.

File: README.md [hunk 4]
```diff
@@ -128,6 +128,30 @@ if (!supportsTime) {
 
 This will add a global `TimePolyfill` function to the page.
 
+### Shadow DOM
```
Comment (Dan503) [inline README.md]: I think this is supposed to be a heading level 2, not heading level 3.

**Review by Dan503 — CHANGES_REQUESTED** [review]
Thanks for also updating the Readme. I've left some comments.

File: README.md (hunk 1 above)
Comment (russcarver) [reply README.md]: It is valid. It's an HTML variable. But I'll remove it when I make the example simpler to not use Angular & TypeScript.

File: README.md [hunk 5]
```diff
@@ -128,6 +128,29 @@ if (!supportsTime) {
 
 This will add a global `TimePolyfill` function to the page.
 
+## Shadow DOM
+
+When your code is inside a component that resides in the Shadow DOM, the polyfill will not be able to find your label
+element. For this case, you can pass your label element in directly.
+
+```
+<label id="myLabel"></label>
```
Comment (Dan503) [inline README.md]: This is an inaccessible label.

Add a `for="timeInput"` attribute to the label to make it accessible.

File: README.md [hunk 6]
```diff
@@ -128,6 +128,29 @@ if (!supportsTime) {
...
+<input type="time" id="timeInput">
+```         
+
+```               
+import * as timePolyFill from 'time-input-polyfill';
```
Comment (Dan503) [inline README.md]: For normal JavaScript users, the import is just this:

```
import timePolyfill from 'time-input-polyfill'
```

The * thing is something only Typescript users have to do.

Comment (Dan503) [reply README.md]: If you want to add a mention of * then it can go up in the area of the readme where I say how to import.

**Review by Dan503 — CHANGES_REQUESTED** [review]
A couple more readme issues.

File: README.md [hunk 7]
```diff
@@ -48,6 +48,9 @@ require('time-input-polyfill/auto');
 
 // ES6
 import 'time-input-polyfill/auto';
+
+// TypeScript
+import * as timePolyFill from 'time-input-polyfill';
```
Comment (Dan503) [inline README.md]: Oh sorry, this isn't actually right.

This section is talking about importing the auto application of the polyfill, not the default polyfill export.

This should just be removed. We don't need to document a Typescript import.

**Review by Dan503 — CHANGES_REQUESTED** [review]

**Review by Dan503 — APPROVED** [review]
Looks good now 🙂

**Comment by Dan503** [comment]
I will need to do some testing before I can merge this into master and publish it.

**Comment by Dan503** [comment]
My weekend is a bit busy. I will aim to publish on Monday night.

**Pull request merged**